stable/nginx-ingress/templates/controller-deployment.yaml

```diff
@@ -10,6 +10,14 @@ metadata:
     release: {{ .Release.Name }}
   name: {{ template "nginx-ingress.controller.fullname" . }}
 spec:
+  selector:
+    matchLabels:
+      app: {{ template "nginx-ingress.name" . }}
+      component: "{{ .Values.controller.name }}"
+      release: {{ .Release.Name }}
+      {{- if .Values.controller.podLabels }}
+{{ toYaml .Values.controller.podLabels | indent 6 }}
```

Comment:
Same as previous comment.

> Let's not add the podLabels to the selector since this field is immutable.

Reply:
According to [https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector):

> `.spec.selector` is an required field that specifies a label selector for the Pods targeted by this Deployment.
>
> `.spec.selector` must match `.spec.template.metadata.labels`, or it will be rejected by the API.
> 
> In API version `apps/v1`, `.spec.selector` and `.metadata.labels` do not default to `.spec.template.metadata.labels` if not set. So they must be set explicitly. Also note that `.spec.selector` is immutable after creation of the Deployment in `apps/v1`.